client/foundries.go

```diff
@@ -1193,3 +1214,89 @@ func (a *Api) FactoryPatchCA(factory string, certs CaCerts) error {
 	_, err = a.Patch(url, data)
 	return err
 }
+
+func (a *Api) FactoryCreateWave(factory string, wave *WaveCreate) error {
+	url := a.serverUrl + "/ota/factories/" + factory + "/waves/"
+	logrus.Debugf("Creating factory wave %s", url)
+
+	data, err := json.Marshal(wave)
+	if err != nil {
+		return err
+	}
+
+	_, err = a.Post(url, data)
+	return err
+}
+
+func (a *Api) FactoryListWaves(factory string) ([]Wave, error) {
+	url := a.serverUrl + "/ota/factories/" + factory + "/waves/"
+	logrus.Debugf("Listing factory waves %s", url)
+
+	body, err := a.Get(url)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I'm wondering if we should make this a paginated API?

Reply:
We don't have a pagination for targets API, and we're going to have less waves than targets.
Thus I thought it's not worth to introduce a pagination into this API.

Though we might make a `fioctl` command paginated... I'm inclined to a `git log -n N` mode which shows only N recent waves, we could set it to 20 by default and use `--all` to show all waves.  An API can stay without pagination until we have a customer creating 1000 waves.

Reply:
targets is hard to paginate since its format is dictated by TUF. Its likely that over time you'll have more waves than targets. Old targets will get pruned so that the production targets is only ever a few versions. 

We do the "-n" type thing for `config log`.  

I don't like the complexity pagination adds, but it does limit the damage that can be caused in a single API call. And its hard to "backport" pagination.

Reply:
Will add an ability to limit a number of returned rows; but probably not a pagination (i.e. just limit, but not offset).
I don't see a use case when we'd need an offset for waves, whereas it adds up to complexity.

Reply:
don't bother. either paginate or do nothing. something in between isn't worth the extra code. the case for pagination is when you've done something like 100 waves. Its a lot to return in a single API call.

Reply:
Done in a `device log` fashion; imho that's enough